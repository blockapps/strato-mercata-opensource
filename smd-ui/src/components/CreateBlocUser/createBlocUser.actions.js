export const OPEN_OVERLAY = "OPEN_MODAL";
export const CLOSE_OVERLAY = "CLOSE_MODAL";
export const CREATE_BLOC_USER_REQUEST = "CREATE_BLOC_USER_REQUEST";
export const CREATE_BLOC_USER_SUCCESS = "CREATE_BLOC_USER_SUCCESS";
export const CREATE_BLOC_USER_FAILURE = "CREATE_BLOC_USER_FAILURE";

export const openOverlay = function () {
  return {
    type: OPEN_OVERLAY,
    isOpen: true
  }
}

export const closeOverlay = function () {
  return {
    type: CLOSE_OVERLAY,
    isOpen: false
  }
}

export const createBlocUser = function (username, password) {
  return {
    type: CREATE_BLOC_USER_REQUEST,
    username,
    password,
    spinning: true,
    isOpen: true,
  }
}

export const createBlocUserSuccess = function (key) {
  return {
    type: CREATE_BLOC_USER_SUCCESS,
    key: key,
    spinning: false,
    isOpen: false,
  }
}

export const createBlocUserFailure = function (error) {
  return {
    type: CREATE_BLOC_USER_FAILURE,
    error: error,
    spinning: false,
    isOpen: false,
  }
}