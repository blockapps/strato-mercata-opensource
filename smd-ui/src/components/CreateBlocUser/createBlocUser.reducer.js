import {
  OPEN_OVERLAY,
  CLOSE_OVERLAY,
  CREATE_BLOC_USER_REQUEST,
  CREATE_BLOC_USER_FAILURE,
  CREATE_BLOC_USER_SUCCESS,
} from './createBlocUser.actions';

const initialState = {
  isOpen: false,
  spinning: false,
  key: null
};

const reducer = function (state = initialState, action) {
  switch (action.type) {
    case OPEN_OVERLAY:
      return {
        isOpen: true,
      };
    case CLOSE_OVERLAY:
      return {
        isOpen: false
      };
    case CREATE_BLOC_USER_REQUEST:
      return {
        isOpen: true,
        spinning: true,
      };
    case CREATE_BLOC_USER_FAILURE:
      return {
        isOpen: false,
        spinning: false,
        error: action.error
      };
    case CREATE_BLOC_USER_SUCCESS:
      return {
        isOpen: false,
        spinning: false,
        key: action.key,
      };
    default:
      return state;
  }
};

export default reducer;