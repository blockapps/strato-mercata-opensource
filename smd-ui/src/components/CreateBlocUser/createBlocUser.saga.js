import {
  takeLatest,
  put,
  call
} from 'redux-saga/effects';
import {
  CREATE_BLOC_USER_REQUEST,
  createBlocUserSuccess,
  createBlocUserFailure,
} from './createBlocUser.actions';

import {
  fetchAccounts
} from '../Accounts/accounts.actions';

import { env } from '../../env';
import { handleErrors } from '../../lib/handleErrors';
import { createUrl } from '../../lib/url';

export function createBlocUserApiCall(username, password) {
  const options = { params: { username } };
  const url = env.BLOC_URL + createUrl("/users/::username", options);

  return fetch(
    url,
    {
      method: 'POST',
      credentials: "include",
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(password)
    }
  )
    .then(handleErrors)
    .then(function (response) {
      return response;
    })
    .catch(function (error) {
      throw error;
    });
}

export function* createBlocUser(action) {
  try {
    let response = yield call(createBlocUserApiCall, action.username, action.password);
    yield put(createBlocUserSuccess(response));
    yield put(fetchAccounts(false, false));
  }
  catch (err) {
    yield put(createBlocUserFailure(err));
  }
}

export default function* watchCreateBlocUser() {
  yield takeLatest(CREATE_BLOC_USER_REQUEST, createBlocUser);
}
