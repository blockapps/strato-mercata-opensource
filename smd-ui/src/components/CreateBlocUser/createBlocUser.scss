.input {
  padding: 1%;
  margin-top: 1%;
  margin-left: 10%;
}

.bloc-user-form .form-width {
  width: 100%;
}