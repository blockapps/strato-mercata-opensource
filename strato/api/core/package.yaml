name:                core-api2

dependencies:
- base
- persistent-postgresql
- servant-server
- strato-conf


library:
  source-dirs: src
  dependencies:
  - aeson
  - aeson-casing
  - base64-bytestring
  - binary
  - blockapps-data
  - blockapps-datadefs
  - bytestring
  - clock
  - common-log
  - conduit
  - containers
  - deepseq
  - directory
  - esqueleto
  - ethereum-discovery
  - extra
  - filepath
  - format
  - hflags
  - http-types
  - kafka-monad
  - labeled-error
  - lens
  - monad-alter
  - mtl
  - process
  - QuickCheck
  - seqevents
  - servant-client
  - servant-multipart
  - servant-multipart-client  
#  - servant-swagger-tags
  - source-tools
  - sql-monad
  - strato-model
  - swagger2
  - template-haskell
  - text
  - transformers
  - type-lits
  - unliftio
  - unordered-containers
  - blockapps-vault-wrapper-api
  - blockapps-vault-wrapper-client
  - blockapps-identity-provider-client
  - vault-monad
  - identity-monad
  - strato-redis-blockdb
  - evm-solidity
