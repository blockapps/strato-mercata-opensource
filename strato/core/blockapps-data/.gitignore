genesis.json
*~
dist
.DS_Store
#*#
blockapps-data.cabal
