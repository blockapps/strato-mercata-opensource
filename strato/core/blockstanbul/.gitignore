-- experimental hpack trial
blockstanbul.cabal
