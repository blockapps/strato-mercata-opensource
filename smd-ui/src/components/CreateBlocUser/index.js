import React, { Component } from 'react';
import { openOverlay, closeOverlay, createBlocUser } from './createBlocUser.actions';
import { Button, Dialog, Intent } from '@blueprintjs/core';
import { Field, reduxForm } from 'redux-form';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

import './createBlocUser.css';
import mixpanelWrapper from '../../lib/mixpanelWrapper';

class CreateBlocUser extends Component {

  componentDidMount() {
    mixpanelWrapper.track("create_user_loaded");
  }

  submit = (values) => {
    mixpanelWrapper.track('create_user_submit_click');
    this.props.createBlocUser(values.username, values.password);
  }

  render() {
    return (
    <div className="smd-pad-16">
      <Button onClick={() => {
        mixpanelWrapper.track('create_user_open_click');
        this.props.reset();
        this.props.openOverlay();
      }} className="pt-intent-primary pt-icon-add"
              id="accounts-create-user-button"
              text="Create User"/>


      <Dialog
          iconName="inbox"
          isOpen={this.props.isOpen}
          onClose={this.props.closeOverlay}
          title="Create New User"
          className="pt-dark"
      >
        <form>
          <div className="pt-dialog-body bloc-user-form">
            <div className="pt-form-group">
              <div className="pt-form-group pt-intent-danger">
                <label className="pt-label" htmlFor="input-a">
                  Username
                </label>
                <div className="pt-form-content">
                  <Field
                      name="username"
                      component="input"
                      type="text"
                      placeholder="Username"
                      className="pt-input form-width"
                      tabIndex="1"
                      required
                  />
                  <div className="pt-form-helper-text">{this.props.errors && this.props.errors.username}</div>
                </div>
              </div>

              <div className="pt-form-group pt-intent-danger">
                <label className="pt-label" htmlFor="input-b">
                  Password
                </label>
                <div className="pt-form-content">
                  <Field
                      name="password"
                      component="input"
                      type="password"
                      placeholder="Password"
                      className="pt-input form-width"
                      tabIndex="2"
                      required
                  />
                  <div className="pt-form-helper-text">{this.props.errors && this.props.errors.password}</div>
                </div>
              </div>

              <div className="pt-form-group pt-intent-danger">
                <label className="pt-label" htmlFor="input-b">
                  Confirm Password
                </label>
                <div className="pt-form-content">
                  <Field
                      name="confirm_password"
                      component="input"
                      type="password"
                      placeholder="Confirm Password"
                      className="pt-input form-width"
                      tabIndex="3"
                      required
                  />
                  <div className="pt-form-helper-text">{this.props.errors && this.props.errors.confirm_password}</div>
                </div>
              </div>
            </div>

            <div>
              <div className="col-sm-3"></div>
              <div className="col-sm-3"></div>
              <div className="col-sm-3"></div>
            </div>
          </div>

          <div className="pt-dialog-footer">
            <div className="pt-dialog-footer-actions">
              <Button text="Cancel" onClick={() => {
                mixpanelWrapper.track('create_user_close_click');
                this.props.reset();
                this.props.closeOverlay();
              }}/>
              <Button
                  intent={Intent.PRIMARY}
                  onClick={this.props.handleSubmit(this.submit)}
                  text="Create User"
              />
            </div>
          </div>
        </form>
      </Dialog>

    </div>
    );
  }
}

export function mapStateToProps(state) {
  let errors = { errors: undefined };
  if (state.form && state.form["create-bloc-user"]) {
    errors = { errors: state.form["create-bloc-user"].syncErrors }
  }
  return {
    isOpen: state.createBlocUser.isOpen,
    ...errors
  };
}

export function validate(values) {
  const errors = {};
  if (!values.username) {
    errors.username = "Username Required";
  }
  if (!values.password) {
    errors.password = "Password Required";
  }
  if (!values.confirm_password) {
    errors.confirm_password = "Must Confirm Password";
  }
  if (values.password !== values.confirm_password) {
    errors.confirm_password = "Passwords Do Not Match";
  }
  return errors;
}

const formed = reduxForm({ form: 'create-bloc-user', validate })(CreateBlocUser);
const connected = connect(
  mapStateToProps,
  {
    openOverlay,
    closeOverlay,
    createBlocUser,
  }
)(formed);

export default withRouter(connected);
