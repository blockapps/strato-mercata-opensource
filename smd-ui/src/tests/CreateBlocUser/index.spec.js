import React from 'react';
import CreateBlocUser, { mapStateToProps, validate } from '../../components/CreateBlocUser';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { Dialog } from '@blueprintjs/core';
import { reducer as formReducer } from 'redux-form'
import { createStore, combineReducers } from 'redux'
import ReactDOM from 'react-dom';

describe('createBlocUser: index', () => {
  let store

  beforeEach(() => {
    store = createStore(combineReducers({ form: formReducer }));
  });

  describe('render component', () => {

    test('without values', () => {
      const props = {
        filter: '',
        contracts: {},
        errors: {},
        openOverlay: jest.fn(),
        closeOverlay: jest.fn(),
        createBlocUser: jest.fn(),
        store: store
      }
      const wrapper = shallow(
        <CreateBlocUser.WrappedComponent {...props} />
      ).dive().dive().dive();

      expect(wrapper.debug()).toMatchSnapshot();
    });

    test('with values', () => {
      const props = {
        filter: '',
        contracts: {},
        errors: {
          confirm_password: "Must Confirm Password",
          password: "Password Required",
          username: "Username Required"
        },
        openOverlay: jest.fn(),
        closeOverlay: jest.fn(),
        createBlocUser: jest.fn(),
        store: store
      }
      const wrapper = shallow(
        <CreateBlocUser.WrappedComponent {...props} />
      ).dive().dive().dive();

      expect(wrapper.debug()).toMatchSnapshot();
    });

  });

  describe('simulate', () => {

    test('open modal', () => {
      const props = {
        filter: '',
        contracts: {},
        openOverlay: jest.fn(),
        closeOverlay: jest.fn(),
        createBlocUser: jest.fn()
      }
      const wrapper = mount(
        <Provider store={store}>
          <CreateBlocUser.WrappedComponent {...props} />
        </Provider>
      );
      wrapper.find('Button').simulate('click');
      expect(props.openOverlay).toHaveBeenCalled();
    });

    test('close modal on outside click', () => {
      const props = {
        filter: '',
        contracts: {},
        isOpen: true,
        openOverlay: jest.fn(),
        closeOverlay: jest.fn(),
        createBlocUser: jest.fn()
      }
      const wrapper = mount(
        <Provider store={store}>
          <CreateBlocUser.WrappedComponent {...props} />
        </Provider>
      );
      wrapper.find(Dialog).get(0).props.onClose();
      expect(props.closeOverlay).toHaveBeenCalled();
    });

    test('close modal on button click', () => {
      const props = {
        filter: '',
        contracts: {},
        isOpen: true,
        openOverlay: jest.fn(),
        closeOverlay: jest.fn(),
        createBlocUser: jest.fn(),
        store: store
      }
      const wrapper = shallow(
        <CreateBlocUser.WrappedComponent {...props} />
      ).dive().dive().dive();
      wrapper.find('Button').at(1).simulate('click')
      expect(props.closeOverlay).toHaveBeenCalled();
    });

    test('submit form', () => {
      const props = {
        filter: '',
        contracts: {},
        isOpen: true,
        openOverlay: jest.fn(),
        closeOverlay: jest.fn(),
        createBlocUser: jest.fn(),
        store: store
      }
      const wrapper = shallow(
        <CreateBlocUser.WrappedComponent {...props} />
      ).dive().dive().dive();
      wrapper.find('Button').at(2).simulate('click')
      expect(props.createBlocUser).toHaveBeenCalled()
    });

  })

  test('component methods', () => {
    const props = {
      filter: '',
      contracts: {},
      openOverlay: jest.fn().mockReturnValue('Open'),
      closeOverlay: jest.fn().mockReturnValue('Close'),
      createBlocUser: jest.fn().mockReturnValue('Create')
    }
    const wrapper = shallow(
      <Provider store={store}>
        <CreateBlocUser.WrappedComponent {...props} />
      </Provider>
    ).dive();

    expect(wrapper.instance().props.openOverlay()).toBe('Open');
    expect(wrapper.instance().props.closeOverlay()).toBe('Close');
    expect(wrapper.instance().props.createBlocUser()).toBe('Create');
  });

  describe('mapStateToProps', () => {

    test('with errors', () => {
      const state = {
        "form": {
          "create-user": {
            "syncErrors": {
              confirm_password: "Must Confirm Password",
              password: "Password Required",
              username: "Username Required"
            }
          }
        },
        "createBlocUser": {
          isOpen: true
        }
      }
      expect(mapStateToProps(state)).toMatchSnapshot();
    });

    test('without errors', () => {
      const state = {
        "form": {},
        "createBlocUser": {
          isOpen: true
        }
      }
      expect(mapStateToProps(state)).toMatchSnapshot();
    });

  });

  describe('validate', () => {

    test('with values', () => {
      const values = {
        confirm_password: "pass",
        password: "pass",
        username: "tanuj"
      }
      expect(validate(values)).toMatchSnapshot();
    });

    test('with empty values', () => {
      const values = {
        confirm_password: null,
        password: null,
        username: null
      }
      expect(validate(values)).toMatchSnapshot();
    });

    test('when password does not match', () => {
      const values = {
        confirm_password: "pass",
        password: "pas",
        username: "tanuj"
      }
      expect(validate(values)).toMatchSnapshot();
    });

  });

});