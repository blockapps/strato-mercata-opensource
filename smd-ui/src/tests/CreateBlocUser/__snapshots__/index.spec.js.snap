// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createBlocUser: index mapStateToProps with errors 1`] = `
Object {
  "errors": undefined,
  "isOpen": true,
}
`;

exports[`createBlocUser: index mapStateToProps without errors 1`] = `
Object {
  "errors": undefined,
  "isOpen": true,
}
`;

exports[`createBlocUser: index render component with values 1`] = `
"<div className=\\"smd-pad-16\\">
  <Blueprint.Button onClick={[Function: onClick]} className=\\"pt-intent-primary pt-icon-add\\" id=\\"accounts-create-user-button\\" text=\\"Create User\\" />
  <Blueprint.Dialog iconName=\\"inbox\\" isOpen={false} onClose={[Function: mockConstructor]} title=\\"Create New User\\" className=\\"pt-dark\\" canOutsideClickClose={true}>
    <form>
      <div className=\\"pt-dialog-body bloc-user-form\\">
        <div className=\\"pt-form-group\\">
          <div className=\\"pt-form-group pt-intent-danger\\">
            <label className=\\"pt-label\\" htmlFor=\\"input-a\\">
              Username
            </label>
            <div className=\\"pt-form-content\\">
              <Field name=\\"username\\" component=\\"input\\" type=\\"text\\" placeholder=\\"Username\\" className=\\"pt-input form-width\\" tabIndex=\\"1\\" required={true} />
              <div className=\\"pt-form-helper-text\\">
                Username Required
              </div>
            </div>
          </div>
          <div className=\\"pt-form-group pt-intent-danger\\">
            <label className=\\"pt-label\\" htmlFor=\\"input-b\\">
              Password
            </label>
            <div className=\\"pt-form-content\\">
              <Field name=\\"password\\" component=\\"input\\" type=\\"password\\" placeholder=\\"Password\\" className=\\"pt-input form-width\\" tabIndex=\\"2\\" required={true} />
              <div className=\\"pt-form-helper-text\\">
                Password Required
              </div>
            </div>
          </div>
          <div className=\\"pt-form-group pt-intent-danger\\">
            <label className=\\"pt-label\\" htmlFor=\\"input-b\\">
              Confirm Password
            </label>
            <div className=\\"pt-form-content\\">
              <Field name=\\"confirm_password\\" component=\\"input\\" type=\\"password\\" placeholder=\\"Confirm Password\\" className=\\"pt-input form-width\\" tabIndex=\\"3\\" required={true} />
              <div className=\\"pt-form-helper-text\\">
                Must Confirm Password
              </div>
            </div>
          </div>
        </div>
        <div>
          <div className=\\"col-sm-3\\" />
          <div className=\\"col-sm-3\\" />
          <div className=\\"col-sm-3\\" />
        </div>
      </div>
      <div className=\\"pt-dialog-footer\\">
        <div className=\\"pt-dialog-footer-actions\\">
          <Blueprint.Button text=\\"Cancel\\" onClick={[Function: onClick]} />
          <Blueprint.Button intent={0} onClick={[Function]} text=\\"Create User\\" />
        </div>
      </div>
    </form>
  </Blueprint.Dialog>
</div>"
`;

exports[`createBlocUser: index render component without values 1`] = `
"<div className=\\"smd-pad-16\\">
  <Blueprint.Button onClick={[Function: onClick]} className=\\"pt-intent-primary pt-icon-add\\" id=\\"accounts-create-user-button\\" text=\\"Create User\\" />
  <Blueprint.Dialog iconName=\\"inbox\\" isOpen={false} onClose={[Function: mockConstructor]} title=\\"Create New User\\" className=\\"pt-dark\\" canOutsideClickClose={true}>
    <form>
      <div className=\\"pt-dialog-body bloc-user-form\\">
        <div className=\\"pt-form-group\\">
          <div className=\\"pt-form-group pt-intent-danger\\">
            <label className=\\"pt-label\\" htmlFor=\\"input-a\\">
              Username
            </label>
            <div className=\\"pt-form-content\\">
              <Field name=\\"username\\" component=\\"input\\" type=\\"text\\" placeholder=\\"Username\\" className=\\"pt-input form-width\\" tabIndex=\\"1\\" required={true} />
              <div className=\\"pt-form-helper-text\\" />
            </div>
          </div>
          <div className=\\"pt-form-group pt-intent-danger\\">
            <label className=\\"pt-label\\" htmlFor=\\"input-b\\">
              Password
            </label>
            <div className=\\"pt-form-content\\">
              <Field name=\\"password\\" component=\\"input\\" type=\\"password\\" placeholder=\\"Password\\" className=\\"pt-input form-width\\" tabIndex=\\"2\\" required={true} />
              <div className=\\"pt-form-helper-text\\" />
            </div>
          </div>
          <div className=\\"pt-form-group pt-intent-danger\\">
            <label className=\\"pt-label\\" htmlFor=\\"input-b\\">
              Confirm Password
            </label>
            <div className=\\"pt-form-content\\">
              <Field name=\\"confirm_password\\" component=\\"input\\" type=\\"password\\" placeholder=\\"Confirm Password\\" className=\\"pt-input form-width\\" tabIndex=\\"3\\" required={true} />
              <div className=\\"pt-form-helper-text\\" />
            </div>
          </div>
        </div>
        <div>
          <div className=\\"col-sm-3\\" />
          <div className=\\"col-sm-3\\" />
          <div className=\\"col-sm-3\\" />
        </div>
      </div>
      <div className=\\"pt-dialog-footer\\">
        <div className=\\"pt-dialog-footer-actions\\">
          <Blueprint.Button text=\\"Cancel\\" onClick={[Function: onClick]} />
          <Blueprint.Button intent={0} onClick={[Function]} text=\\"Create User\\" />
        </div>
      </div>
    </form>
  </Blueprint.Dialog>
</div>"
`;

exports[`createBlocUser: index validate when password does not match 1`] = `
Object {
  "confirm_password": "Passwords Do Not Match",
}
`;

exports[`createBlocUser: index validate with empty values 1`] = `
Object {
  "confirm_password": "Must Confirm Password",
  "password": "Password Required",
  "username": "Username Required",
}
`;

exports[`createBlocUser: index validate with values 1`] = `Object {}`;
